### PR TITLE
docker info suppports case-insensitive proxy env settings

### PR DESCRIPTION
**Pull request by nak3**
Signed-off-by: Kenjiro Nakayama nakayamakenjiro@gmail.com

### CODE REVIEW

File: daemon/info.go [hunk 1]
```diff
@@ -111,3 +112,12 @@ func (daemon *Daemon) SystemInfo() (*types.Info, error) {
 
 	return v, nil
 }
+
+// HTTP_PROXY, HTTPS_RPOXY and NO_PROXY are case-insensitive.
+// See the Go specification for details on these variables. https://golang.org/pkg/net/http/
+func getProxyEnv(key string) string {
+	if os.Getenv(key) == "" {
+		return os.Getenv(strings.ToUpper(key))
```
Comment (phemmer) [inline daemon/info.go]: The way net/http behaves is that the uppercase variants are tried first.
Also, this function relies on the input being lowercase. If you call `getProxyEnv("HTTP_PROXY")`, the lowercase variant won't be tried.
Basically, should explicitly call `ToUpper` on the first try, and `ToLower` on the second.

File: daemon/info.go [hunk 2]
```diff
@@ -111,3 +112,12 @@ func (daemon *Daemon) SystemInfo() (*types.Info, error) {
 
 	return v, nil
 }
+
+// HTTP_PROXY, HTTPS_RPOXY and NO_PROXY are case-insensitive.
```
Comment (phemmer) [inline daemon/info.go]: Not quite accurate. They can be either all upper, or all lower. `HTTP_proxy` won't work for example.

**Comment by duglin** [comment]
I'm not sure this is the correct solution.  Simply looking at the upper case version of this vars means that we'll ignore the lower-case version when only the lower-case one is set.  We should mimic what golang does, which I believe is to look at the upper-case first, then the lower-case one.

**Comment by nak3** [comment]
Thank you for your review. How about this one?

File: daemon/info.go [hunk 3]
```diff
@@ -111,3 +112,13 @@ func (daemon *Daemon) SystemInfo() (*types.Info, error) {
 
 	return v, nil
 }
+
+// The uppercase and the lowercase are available for the proxy settings.
+// See the Go specification for details on these variables. https://golang.org/pkg/net/http/
+func getProxyEnv(key string) string {
+	proxyValue := os.Getenv(strings.ToUpper(key))
+	if proxyValue == "" {
+		return os.Getenv(key)
```
Comment (phemmer) [inline daemon/info.go]: Still relies on the case being passed to the function. If you call `getProxyEnv("FOO")`, it won't pick up the env var `foo`.

Comment (nak3) [inline daemon/info.go]: @phemmer I mentioned it in docs/reference/commandline/cli.md
https://github.com/nak3/docker/commit/8b1fc7f59957473a0971bfcf93226896378a6317

Comment (phemmer) [inline daemon/info.go]: That doesn't mention anything about it. It also wouldn't belong there as the behavior is in regards to the usage of the `getProxyEnv` function, which a user of the CLI has no interaction with.

Comment (nak3) [inline daemon/info.go]: @phemmer Sorry, I misunderstood that. I updated,

**Comment by crosbymichael** [comment]
LGTM

@nak3 can you rebase?

**Comment by nak3** [comment]
@crosbymichael Thank you. I rebased.

**Comment by calavera** [comment]
LGTM

**Pull request merged**